go/worker/compute/executor/committee/node.go

```diff
@@ -729,25 +730,40 @@ func (n *Node) proposeTimeoutLocked(roundCtx context.Context) error {
 	return nil
 }
 
-func (n *Node) getRtStateAndRoundResults(ctx context.Context, height int64) (*roothash.RuntimeState, *roothash.RoundResults, error) {
+func (n *Node) getRuntimeRoundData(ctx context.Context, height int64, currentRound uint64) (*roothash.RuntimeState, *roothash.RoundResults, []*staking.Event, error) {
 	state, err := n.commonNode.Consensus.RootHash().GetRuntimeState(ctx, n.commonNode.Runtime.ID(), height)
 	if err != nil {
 		n.logger.Error("failed to query runtime state",
 			"err", err,
 			"height", height,
 		)
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	roundResults, err := n.commonNode.Runtime.History().GetRoundResults(ctx, state.LastNormalRound)
 	if err != nil {
 		n.logger.Error("failed to query round last normal round results",
 			"err", err,
 			"height", height,
 		)
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	return state, roundResults, nil
+	// Load all events since last successful round.
```

Comment:
This can be a lot of events in theory. Especially if the runtime is suspended for a while.

Reply:
Yeah indeed. Not sure what the alternative is, limit max number of events? But in that case the runtime cannot really rely on getting all of the events?

Reply:
Maybe we could do it the same as for transactions -- support batching of events and have a max number of events per batch? The downside would be that this could delay runtimes being informed of consensus changes.

Unfortunately I don't see how this could work without consensus breaking changes as you would need to track which events were included in a batch (e.g., up to which height) and this information (together with the events hash) would need to be subject to discrepancy detection.

Reply:
Hm yeah i think we might indeed need batching. Since the alternatives i can think of are: 
- drop events? - then every runtime needs to be aware that events can be dropped, at that point the runtime would just query consensus state instead of follow events
- share the (untrusted) index db store with the runtime, then the runtime would need to compute the events hash of the events used and provide it for dd. But this probably just moves the batching work to the runtime, not really solves anything?
- not limit - in case of a suspended runtime this can grow unlimitedly - making the runtime unable to be resumed again (due to taking to long to process all the events)